Content.Server/Ninja/Systems/NinjaGlovesSystem.cs

```diff
@@ -0,0 +1,304 @@
+using Content.Server.Actions;
+using Content.Server.Chat.Systems;
+using Content.Server.Communications;
+using Content.Server.Doors.Systems;
+using Content.Server.DoAfter;
+using Content.Server.Electrocution;
+using Content.Server.GameTicking;
+using Content.Server.GameTicking.Rules;
+using Content.Server.Popups;
+using Content.Server.Power.Components;
+using Content.Shared.Actions;
+using Content.Shared.Administration.Logs;
+using Content.Shared.Damage.Components;
+using Content.Shared.Database;
+using Content.Shared.DoAfter;
+using Content.Shared.Doors.Components;
+using Content.Shared.Emag.Systems;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction;
+using Content.Shared.Ninja.Components;
+using Content.Shared.Ninja.Systems;
+using Content.Shared.Popups;
+using Content.Shared.Research.Components;
+using Content.Shared.Tag;
+using Robust.Shared.Audio;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Server.Ninja.Systems;
+
+public sealed class NinjaGlovesSystem : SharedNinjaGlovesSystem
+{
+    [Dependency] private readonly ActionsSystem _actions = default!;
+    [Dependency] private readonly ISharedAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly ChatSystem _chat = default!;
+    [Dependency] private readonly DoAfterSystem _doafter = default!;
+    [Dependency] private readonly ElectrocutionSystem _electrocution = default!;
+    [Dependency] private readonly EmagSystem _emag = default!;
+    [Dependency] private readonly GameTicker _gameTicker = default!;
+    [Dependency] private readonly NinjaSystem _ninja = default!;
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly TagSystem _tags = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ActivateInWorldEvent>(OnActivate);
+        SubscribeLocalEvent<NinjaGlovesComponent, DoAfterEvent>(EndedDoAfter);
+        // TODO: raising 1 event triggers for all 3 components sicne on same entity, change to be per-component?
+        SubscribeLocalEvent<NinjaDrainComponent, DoAfterEvent>(OnDrainDoAfter);
+        SubscribeLocalEvent<NinjaDownloadComponent, DoAfterEvent>(OnDownloadDoAfter);
+        SubscribeLocalEvent<NinjaTerrorComponent, DoAfterEvent>(OnTerrorDoAfter);
+
+        SubscribeLocalEvent<DoorComponent, DoorEmaggedEvent>(OnDoorEmagged);
+    }
+
+    protected override void NinjaEquippedGloves(EntityUid uid, NinjaGlovesComponent comp, EntityUid user, NinjaComponent ninja)
+    {
+        base.NinjaEquippedGloves(uid, comp, user, ninja);
+
+        if (TryComp<ActionsComponent>(user, out var actions))
+            _actions.AddAction(user, comp.ToggleAction, uid, actions);
+    }
```

Comment:
```suggestion
    protected override void NinjaEquippedGloves(EntityUid uid, NinjaGlovesComponent comp, EntityUid user, NinjaComponent ninja)
    {
        base.NinjaEquippedGloves(uid, comp, user, ninja);

        _interaction.SetRelay(user, uid, EnsureComp<InteractionRelayComponent>(user));

        if (TryComp<ActionsComponent>(user, out var actions))
            _actions.AddAction(user, comp.ToggleAction, uid, actions);
    }
```

This is to set the relay. Once you change this to the GetItemActionsEvent, move it there.